mbs_results/data_cleaning.py

```diff
@@ -59,4 +60,6 @@ def clean_and_merge(
     responses = responses[responses_keep_cols].set_index([reference, period])
     contributors = contributors[contributors_keep_cols].set_index([reference, period])
 
-    return responses.join(contributors, on=[reference, period])
+    validate_indices(responses, contributors)
+
+    return responses.merge(contributors, on=[reference, period])
```

Comment:
returned data frame still has reference and period as index. Is it worth resetting index before returning to be similar to other function inputs?

Reply:
I think it makes sense to keep it that way - we want those 2 variables as indices for the rest of the pipeline.